File: backend/src/main/java/ru/secretsanta/mapper/user/InviteMapper.java
package ru.secretsanta.mapper.user;

import ru.secretsanta.dto.response.InviteResponse;
import ru.secretsanta.entity.user.InviteToken;

public class InviteMapper {

    public static InviteResponse toInviteResponse(InviteToken inviteToken) {
        return new InviteResponse(inviteToken.getUsername(), inviteToken.getToken());
    }
}

File: backend/src/main/kotlin/ru/secretsanta/mapper/user/InviteMapper.kt
package ru.secretsanta.mapper.user

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.stereotype.Component
import ru.secretsanta.dto.response.InviteResponse
import ru.secretsanta.entity.user.InviteToken

@Component
@EnableConfigurationProperties(InviteProperties::class)
class InviteMapper(
    private var properties: InviteProperties
) {
    fun toInviteResponse(inviteToken: InviteToken): InviteResponse {
        return InviteResponse(inviteToken.username, properties.host + "/register?token=" + inviteToken.token)
    }
}

@ConfigurationProperties(prefix = "invites")
data class InviteProperties(
     var host: String = "")
